tests/d3d12_depth_stencil.c

```diff
@@ -2409,3 +2409,176 @@ void test_depth_bias_behaviour(void)
 
     destroy_test_context(&context);
 }
+
+void test_depth_bias_formats(void)
+{
+    D3D12_FEATURE_DATA_D3D12_OPTIONS16 options16;
+    D3D12_GRAPHICS_PIPELINE_STATE_DESC pso_desc;
+    ID3D12GraphicsCommandList9 *command_list9;
+    D3D12_DEPTH_STENCIL_VIEW_DESC dsv_desc;
+    D3D12_HEAP_PROPERTIES heap_properties;
+    D3D12_CPU_DESCRIPTOR_HANDLE dsv;
+    ID3D12DescriptorHeap *dsv_heap;
+    struct test_context_desc desc;
+    D3D12_RESOURCE_DESC ds_desc;
+    struct test_context context;
+    ID3D12Resource *ds_resource;
+    ID3D12PipelineState *pso;
+    float clear_depth;
+    unsigned int i, j;
+    HRESULT hr;
+
+#include "shaders/depth_stencil/headers/vs_constant_depth.h"
+
+    static const float red[] = { 1.0f, 0.0f, 0.0f, 1.0f };
+
+    static const struct
+    {
+        DXGI_FORMAT dsv_format;
+        uint32_t depth_clear;
+        uint32_t depth_render;
+        int32_t depth_bias;
+        bool expect_pass;
+    }
+    tests[] =
+    {
+        { DXGI_FORMAT_D16_UNORM,            0x3c802000, 0x3c800000, 0, false },
+        { DXGI_FORMAT_D16_UNORM,            0x3c802000, 0x3c800000, 1, true  },
+        { DXGI_FORMAT_D16_UNORM,            0x3c820000, 0x3c800000, 1, false },
+
+        { DXGI_FORMAT_D24_UNORM_S8_UINT,    0x3c800020, 0x3c800000, 0, false },
+        { DXGI_FORMAT_D24_UNORM_S8_UINT,    0x3c800020, 0x3c800000, 1, true  },
```

Comment:
Isn't this slightly above 1.0? How does that work?

Reply:
To properly test D24_UNORM behavior on emulated format, shouldn't we use smaller FP values here?

Reply:
The numbers are all around 1/64 actually. Will add a comment.

Reply:
Ah, right. I looked at the wrong thing in the FP32 calculator :frog: